README.md

```diff
@@ -1,18 +1,334 @@
+
+Let's say I needed to declare and define multiple variables. It feels like a
+lot to have to repeat `var` over and over again. JavaScript allows us to do
+multi-line variable assignment to alleviate this pain. Every variable must be
+separated with a comma, and the entire line must end with a semicolon.
+
+Let's condense the below code into one line:
+
+```javascript
+var a = 5;
+var b = 2;
+var c = 3;
+var d = {};
+var e = [];
+```
+
+The above is equivalent to:
+
+```javascript
+var a = 5,
+    b = 2,
+    c = 3,
+    d = {},
+    e = [];
+```
+
+which can be converted to:
+
+```javascript
+var a = 5, b = 2, c = 3, d = {}, e = [];
+```
+
+### Retrieving Variables
+
+To retrieve a declared variable, refer to its name:
+```js
+pi;
+//=> 3.14159
+```
+
+### Reassigning Variable Value
+
+Changing the value of a variable in JavaScript works just in the same way
+as it does in Ruby:
+
+```js
+var pi = 3.14159;
+pi;
+//=> 3.14159
+pi = 3.14;
+pi;
+//=> 3.14;
+```
+
+### Variable Values
+
+Upon declaration, all variables are automatically assigned the value of
+`undefined`. It's only after we assign a new value that the variable will
+contain something other than undefined. We can use the `typeof` operator to
+check the data type of the value currently stored in a variable:
+
+```js
+var language;
+//=> undefined
+
+typeof language;
+//=> "undefined"
+
+language = "JavaScript";
+//=> "JavaScript"
+
+typeof language;
+//=> "string"
+```
+
+***Top Tip***: When writing JavaScript code, it's good practice to ***never***
+set a variable equal to `undefined`. Variables will be `undefined` until we
+explicitly assign a value, so encountering an `undefined` variable is a strong
+signal that the variable was declared but not assigned prior to the reference.
+That's valuable information that we can use while debugging, and it comes at no
+additional cost to us.
+
+Once a variable has been created with `var`, we can reassign it to our heart's
+content:
+
+```js
+var pi = 3.14159;
+//=> undefined
+
+typeof pi;
+//=> "number"
+
+pi = "the ratio between a circle's circumference and diameter";
+//=> "the ratio between a circle's circumference and diameter"
+
+typeof pi;
+//=> "string"
+```
+
+The data that's stored in our variable might change over time, but at any moment
+we can retrieve its current contents:
+
+```js
+var language = "Mocha";
+//=> undefined
+
+language = "LiveScript";
+//=> "LiveScript";
+
+language = "JavaScript";
+//=> "JavaScript";
+
+language;
+//=> "JavaScript";
+```
+
+## Identify When to Use `const`, `let`, and `var` for Declaring Variables
+
+Because of its ubiquity in legacy code and StackOverflow posts, it's important
+to get to know `var`. However, as we alluded to earlier, there is almost no
+reason to use `var` with the features JavaScript has post-2015. `var` comes with
+a ton of baggage in the form of scope issues (which we will discuss in the
+lesson on scope in JavaScript) and allows developers to play a little too
```

Comment:
Similar to the Ruby note, let's remove references to lessons not in the same section/brick. JS scope is in section 3.